### PR TITLE
Make clean task work with click

### LINKED ISSUE #5542: [CT-918] `dbt clean` works in click

### PR DESCRIPTION
**Pull request by stu-k**
resolves #5542

### Description
- `dbt clean` works with `click` CLI
- Moved unnecessary methods from `CleanTask` into standalone functions

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/task/clean.py [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
-        protected_abs_paths = [os.path.abspath(p) for p in protected_paths]
-        return abs_path in set(protected_abs_paths) or self.__is_project_path(abs_path)
 
     def run(self):
         """
         This function takes all the paths in the target file
         and cleans the project paths that are not protected.
         """
-        move_to_nearest_project_dir(self.args.project_dir)
+        move_to_nearest_project_dir(self.args.PROJECT_DIR)
```
Comment (MichelleArk) [inline core/dbt/task/clean.py:27]: I believe `self.args.project_dir` should work when the `CleanTask` is run both from the old CLI and new click CLI given the backwards-compatibility changes you've made [here](https://github.com/dbt-labs/dbt-core/pull/6369/files#diff-3e5c2add494f2ffa25c42dd0ea6ea0d4cc43875474b110506dff44385adb81acR37)

I believe this should also fix the integration test errors!

Comment (stu-k) [reply core/dbt/task/clean.py:27]: Oh whoops! Forgot to change this back!

File: core/dbt/cli/flags.py [hunk 2]
```diff
@@ -33,6 +33,8 @@ def assign_params(ctx, params_assigned_from_default):
                 # https://docs.python.org/3/library/dataclasses.html#frozen-instances
                 if hasattr(self, param_name):
                     raise Exception(f"Duplicate flag names found in click command: {param_name}")
+                # TODO: remove when lower case params are no longer required
+                object.__setattr__(self, param_name.lower(), param_value)
```
Comment (iknox-fa) [inline core/dbt/cli/flags.py:37]: When #6336 gets merged (in review now) you can remove this line.

File: core/dbt/task/clean.py [hunk 3]
```diff
@@ -1,54 +1,67 @@
...
                 fire_event(ProtectedCleanPath(path=path))
 
         fire_event(FinishedCleanPaths())
+
+    @classmethod
+    def from_project(cls, project: Project, cli_vars: Dict[str, Any]) -> "CleanTask":
+        return cls(cli_vars, project)
+
+    @classmethod
+    def from_args(cls, args) -> "CleanTask":
```
Comment (iknox-fa) [inline core/dbt/task/clean.py:45]: What is this used by?  I don't see it being called anywhere...

File: core/dbt/cli/main.py [hunk 4]
```diff
@@ -129,7 +131,13 @@ def build(ctx, **kwargs):
 def clean(ctx, **kwargs):
     """Delete all folders in the clean-targets list (usually the dbt_packages and target directories.)"""
     flags = Flags()
-    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+    project = ctx.obj["project"]
+
+    task = CleanTask(flags, project)
```
Comment (ChenyuLInx) [inline core/dbt/cli/main.py:136]: Looks clean!

**Review by ChenyuLInx — COMMENTED** [review]
Looks good to me, can probably do a last round of update to remove the line Ian mentioned after his PR is in.

File: core/dbt/task/clean.py (hunk 3 above)
Comment (stu-k) [reply core/dbt/task/clean.py]: Removed `from_args` from `CleanTask`.

**Review by ChenyuLInx — APPROVED** [review]

File: core/dbt/task/clean.py [hunk 5]
```diff
@@ -1,54 +1,54 @@
...
+from dbt.task.base import (
+    BaseTask,
+    move_to_nearest_project_dir,
+)
 
 
 class CleanTask(BaseTask):
-    # Note: CleanTask is the last task that uses UnsetProfileConfig,
-    # and can be deleted once CleanTask no longer requires it.
-    ConfigType = UnsetProfileConfig
```
Comment (MichelleArk) [inline core/dbt/task/clean.py:20]: Could we rip out `UnsetProfileConfig` at this point? 🔥 Or in a follow-up?

File: core/dbt/task/clean.py [hunk 6]
```diff
@@ -1,54 +1,54 @@
...
+            if not is_protected_path(path, self.config.model_paths, self.config.test_paths):
                 shutil.rmtree(path, True)
                 fire_event(ConfirmCleanPath(path=path))
             else:
                 fire_event(ProtectedCleanPath(path=path))
 
         fire_event(FinishedCleanPaths())
+
+
+def is_protected_path(path, model_paths: List[str], test_paths: List[str]) -> bool:
```
Comment (MichelleArk) [inline core/dbt/task/clean.py:43]: missing type annotation on `path`

File: core/dbt/task/clean.py (hunk 5 above)
Comment (stu-k) [reply core/dbt/task/clean.py:20]: I think in a followup would make the most sense!

**Review by iknox-fa — APPROVED** [review]
LGTM!
The only thing I might change is that we really want to get away from using os.path (preferring PathLib) so if you're feeling like learning how that works I'd be double happy to see this PR fix that although it is well out of the scope of the ticket, so don't let it become a blocker to merge by any means.

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**